src/Observations.jl

```diff
@@ -328,26 +333,37 @@ end
 ##### Initializing simulations
 #####
 
-function initialize_simulation!(simulation, observations, time_series_collector, time_index=1)
-    set!(simulation.model, observations, time_index)
+function initialize_forward_run!(simulation, observations, time_series_collector, time_index=1)
+
+    reset!(simulation)
 
     times = observation_times(observations)
     initial_time = times[time_index]
     simulation.model.clock.time = initial_time
-    simulation.model.clock.iteration = 0
-    simulation.model.timestepper.previous_Δt = Inf
-    simulation.initialized = false
 
+    collected_fields = time_series_collector.collected_fields
+    arch = architecture(time_series_collector.grid)
+
+    # Clear potential NaNs from timestepper data.
```

Comment:
```suggestion
    # Clear potential NaNs from timestepper data.
    # Particularly important for Adams-Bashforth timestepping scheme.
    # Oceananigans ≤ v0.71 initializes the Adams-Bashforth scheme with an Euler step by *multiplying* the tendency
    # at time-step n-1 by 0. Because 0 * NaN = NaN, this fails when the tendency at n-1 contains NaNs.
```

Reply:
Let's not be cryptic about it!